src/main/java/org/mockito/internal/util/DefaultMockingDetails.java

```diff
@@ -36,17 +38,25 @@ public boolean isSpy(){
 
     @Override
     public Collection<Invocation> getInvocations() {
-        return getMockHandler(toInspect).getInvocationContainer().getInvocations();
+        return mockHandler().getInvocationContainer().getInvocations();
     }
 
     @Override
-    public Class<?> getMockedType() {
-        return getMockHandler(toInspect).getMockSettings().getTypeToMock();
+    public MockCreationSettings<?> getMockCreationSettings() {
+        return mockHandler().getMockSettings();
     }
 
-    @Override
-    public Set<Class<?>> getExtraInterfaces() {
-        return getMockHandler(toInspect).getMockSettings().getExtraInterfaces();
+    private InternalMockHandler<Object> mockHandler() {
+        assertGoodMock();
```

Comment:
> We cannot have them on constructor at the moment because isMock / isSpy methods permit null / non-mock arguments. 

Good point. That was a suggestion only. But that work too !

Discoverability won't suffer that much. Imho